pkg/cmd/grafana-cli/commands/ls_command_test.go

```diff
@@ -4,67 +4,81 @@ import (
 	"errors"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/commands/commandstest"
 	"github.com/grafana/grafana/pkg/cmd/grafana-cli/services"
-	. "github.com/smartystreets/goconvey/convey"
 )
 
 func TestMissingPath(t *testing.T) {
-	var org = validateLsCommand
+	tests := []struct {
+		description string
+		cliContext  map[string]string
+		ioHelper    *commandstest.FakeIoUtil
+		error       error
+	}{
+		{
+			description: "missing path flag",
+			cliContext:  make(map[string]string),
+			ioHelper:    &commandstest.FakeIoUtil{},
+			error:       errMissingPathFlag,
+		},
+		{
+			description: "not a directory",
+			cliContext:  map[string]string{"pluginsDir": "/var/lib/grafana/plugins/notadir.txt"},
+			ioHelper:    &commandstest.FakeIoUtil{FakeIsDirectory: false},
+			error:       errNotDirectory,
+		},
+	}
 
-	Convey("ls command", t, func() {
-		validateLsCommand = org
+	for _, tc := range tests {
+		t.Run(tc.description, func(t *testing.T) {
+			origIoHelper := services.IoHelper
+			services.IoHelper = tc.ioHelper
+			t.Cleanup(func() {
+				services.IoHelper = origIoHelper
+			})
 
-		Convey("Missing path flag", func() {
-			cmd := Command{}
-			c, err := commandstest.NewCliContext(map[string]string{})
-			So(err, ShouldBeNil)
-			services.IoHelper = &commandstest.FakeIoUtil{}
+			c, err := commandstest.NewCliContext(tc.cliContext)
+			require.NoError(t, err)
 
-			Convey("should return error", func() {
-				err := cmd.lsCommand(c)
-				So(err, ShouldBeError, "missing path flag")
-			})
+			cmd := Command{}
+			err = cmd.lsCommand(c)
+			assert.Truef(t, errors.Is(err, tc.error), "expected '%s', got '%s'", tc.error.Error(), err.Error())
```

Comment:
I wonder if it's better to be explicit in tests and not use `errors.Is`, but instead assert.EqualError? The test should have an exact expected behaviour after all. Writing such assertions is also much easier.

Reply:
I can change that. I think I overapplied [Change-Detector Tests Considered Harmful](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html) here

Reply:
I changed it to `assert.Equal` instead of `assert.EqualError`, since I have access to the precise version of the error and don't have to unpack `.Error()` for the compare.

Reply:
Ah, I didn't know that pattern, but I also prefer that tests don't cause friction to refactoring. I just don't think it's the case here, but rather defining the output :)

Reply:
@sakjur OK :+1: Thanks for taking care of things :)